src/net_processing.cpp

```diff
@@ -2557,8 +2557,10 @@ void PeerManager::ProcessMessage(CNode& pfrom, const std::string& msg_type, CDat
     // BIP155 defines feature negotiation of addrv2 and sendaddrv2, which must happen
     // between VERSION and VERACK.
     if (msg_type == NetMsgType::SENDADDRV2) {
-        if (pfrom.fSuccessfullyConnected) {
+        if (pfrom.fSuccessfullyConnected && !m_chainparams.GetConsensus().signet_blocks) {
             // Disconnect peers that send a SENDADDRV2 message after VERACK.
+            // A temporary exception is made for bootstrapping signet support in v0.21.0.
+            // After 0.21.0, this can be reverted.
             pfrom.fDisconnect = true;
```

Comment:
Behaviour doesn't match description -- per the commit title, this should be:

```c++
    if (pfrom.fSuccessfullyConnected) {
        if (!m_chainparams.GetConsensus().signet_blocks) {
            pfrom.fDisconnect = true;
        }
        return;
}
```

If you want the behaviour that's implemented, should be described as "support sendaddrv2 after verack on signet only for compatibility with earlier bip drafts" or similar.

Reply:
Thanks, good point, which @MarcoFalke also pointed out in https://github.com/bitcoin/bitcoin/pull/20646#issuecomment-744306427. Updated the implementation for signet behavior to be as close as possible to mainnet.